.github/workflows/pull-request.yml

```diff
@@ -23,61 +30,54 @@ jobs:
       - name: Set environment variables
         run: |
           echo "SFDX_DISABLE_AUTOUPDATE=true" >> $GITHUB_ENV
+          echo "SFDX_DISABLE_SOURCE_MEMBER_POLLING=true" >> $GITHUB_ENV
           echo "SFDX_DISABLE_TELEMETRY=true" >> $GITHUB_ENV
 
       - name: Install SFDX CLI and Scanner
         run: |
           npm install sfdx-cli
           node_modules/sfdx-cli/bin/run plugins:install @salesforce/sfdx-scanner
 
-      - name: "Install npm dependencies"
+      - name: Install npm dependencies
         run: npm install
 
-      - name: "Code formatting verification with Prettier"
+      - name: Code formatting verification with Prettier
         run: npm run prettier:verify
 
-      - name: "Apex static analysis"
-        uses: mitchspano/sfdx-scan-pull-request@v0.1.3
+      - name: Apex static analysis
+        uses: mitchspano/sfdx-scan-pull-request@v0.1.7
         with:
           pmdconfig: apex-ruleset.xml
           severity-threshold: 4
         env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GITHUB_TOKEN: ${{ github.token }}
 
-      - name: "Check documentation writing"
+      - name: Check documentation writing
         run: npm run lint:doc
 
-  ci-build:
-    runs-on: ubuntu-latest
-    steps:
-      - name: Set environment variables
-        run: |
-          echo "SFDX_DISABLE_AUTOUPDATE=true" >> $GITHUB_ENV
-          echo "SFDX_DISABLE_SOURCE_MEMBER_POLLING=true" >> $GITHUB_ENV
-          echo "SFDX_DISABLE_TELEMETRY=true" >> $GITHUB_ENV
-
-      - name: Install Salesforce CLI
-        run: npm install sfdx-cli -g
-
-      - name: Checkout
-        uses: actions/checkout@v3
+  integrate:
+    uses: ./.github/workflows/build.yml
+    secrets: inherit
 
-      - name: "Populate auth file with CI_URL secret"
-        shell: bash
-        run: "echo ${{ secrets.CI_URL}} > ./CI_URL.txt"
+  create-package-version:
+    uses: ./.github/workflows/create-package-version.yml
+    needs: [integrate, formatting-and-linting]
+    with:
+      branch: pull-request
+    secrets: inherit
 
-      - name: "Authenticate CI Sandbox"
-        run: "sfdx force:auth:sfdxurl:store -f ./CI_URL.txt -a ci -d"
-
-      - name: "Create scratch org"
-        run: "sfdx force:org:create -f config/project-scratch-def.json -a scratch-org -s -d 1"
-
-      - name: "Push source to scratch org"
-        run: "sfdx force:source:push"
-
-      - name: "Run Apex tests"
-        run: npm run test
+  commit-lint:
```

Comment:
question: are jobs run in parallel here? if not, IMO we should lint before creating any package version

Reply:
`integrate`, `formatting and linting`, `commit lint` and `pr lint` are run in parallel
`create package version` is run after `integrate` and `formatting and linting` are both successful
So we create a package only if the sources build and they respect our format
But we also build a package if the commit message or the pr message does not follow our convention.
We want to know if sources build and pack in this case but we want to block the PR from merging because we need this convention to be enforced to be able to create version automatically